Makefile

```diff
@@ -8,95 +50,240 @@ include $(ROOT)/deps/readies/mk/main
+
+$(TARGET): $(BINDIR)/Makefile
+	@echo Building $(TARGET) ...
+ifneq ($(DRY_RUN),1)
+	$(SHOW)$(MAKE) -C $(BINDIR) $(MAKE_J)
+else
+	@make -C $(BINDIR) $(MAKE_J)
+endif
+
+clean:
+ifeq ($(ALL),1)
+	$(SHOW)rm -rf $(BINROOT) build dist .tox
+else
+	$(SHOW)$(MAKE) -C $(BINDIR) clean
+endif
+
+.PHONY: clean
+
+#----------------------------------------------------------------------------------------------
+
+pybind:
+	$(SHOW)python3 -m poetry build
+
 #----------------------------------------------------------------------------------------------
 
+_CTEST_ARGS=$(CTEST_ARGS)
+_CTEST_ARGS += \
+	--output-on-failure \
+	$(MAKE_J)
+
 ifeq ($(VERBOSE),1)
-CMAKE_FLAGS+=-DCMAKE_VERBOSE_MAKEFILE=on
+_CTEST_ARGS += -V
+endif
+
+ifeq ($(VALGRIND),1)
+_CTEST_ARGS += \
+	-T memcheck \
+	--overwrite MemoryCheckCommandOptions="--leak-check=full"
 endif
 
-build:
-	mkdir -p build
-	cd build; cmake $(CMAKE_FLAGS) ../src
-	-touch build/Makefile
-	make -C build
+unit_test:
+	$(SHOW)cd $(BINDIR)/unit_tests && GTEST_COLOR=1 ctest $(_CTEST_ARGS)
 
-clean:
-ifneq ($(ALL),1)
-	make -C build clean
+#----------------------------------------------------------------------------------------------
+
+flow_test:
+ifneq ($(VIRTUAL_ENV),)
+	$(SHOW)cd tests/flow && python3 -m pytest $(TEST)
 else
-	rm -rf build
+	$(SHOW)$(MAKE) pybind
+	$(SHOW)python3 -m tox -e flowenv
 endif
 
-.PHONY: build clean
+#----------------------------------------------------------------------------------------------
+
+ifneq ($(TEST),)
+RLTEST_ARGS += --test $(TEST) -s
+endif
+ifeq ($(VERBOSE),1)
+RLTEST_ARGS += -v
+endif
+
+mod_test:
+	$(SHOW)cd $(ROOT)/tests/module/flow && \
+		python3 -m RLTest \
+		--module $(BINDIR)/module_tests//memory_test.so \
+		--clear-logs \
+		$(RLTEST_ARGS)
 
 #----------------------------------------------------------------------------------------------
 
-test:
-	./tests/test.sh
+benchmark:
+	$(SHOW)$(BINDIR)/benchmark/bf_benchmark
+
+toxenv:
+ifeq ($(wildcard .tox),)
+	$(SHOW)tox -e flowenv
+endif
+	$(SHOW)bash -c ". ./.tox/flowenv/bin/activate; $$SHELL"
 
-.PHONY: test
+.PHONY: unit_test flow_test mem_test benchmark toxenv
 
 #----------------------------------------------------------------------------------------------
 
-valgrind:
-	./tests/valgrind.sh
+check-format:
+	$(SHOW)./sbin/check-format.sh
+
+format:
+	$(SHOW)FIX=1 ./sbin/check-format.sh
 
-.PHONY: valgrind
+.PHONY: check-format format
 
 #----------------------------------------------------------------------------------------------
 
 platform:
-	@make -C build/platforms build
+	$(SHOW)make -C build/platforms build PACK=1 ARTIFACTS=1
 ifeq ($(PUBLISH),1)
-	@make -C build/platforms publish
+	$(SHOW)make -C build/platforms publish
 endif
 
-format:
-	./clang-format-all.sh
+.PHONY: platform
 
-lint:
-	./clang-check-all.sh
+#----------------------------------------------------------------------------------------------
 
-benchmark:
-	./tests/benchmark.sh
+ifneq ($(wildcard /w/*),)
+SANBOX_ARGS += -v /w:/w
+endif
```

Comment:
what is the purpose of this check?

Reply:
It will mount `/w` on `/w` if such a directory exists. It allows one to mount all their repos on `/w` and get access to them.